### ISSUE TITLE
[explicit-module-boundary-types] Ignore Patterns

### ISSUE THREAD

**Issue by musicMan1337** [typescript-eslint/typescript-eslint#2764]
**Repro**

I'm wondering if an ignore pattern option can be added, particularly in the case of React Components. Naming convention is to capitalize the first letter, and I'd like to not have to type the return on components since it can be somewhat tedious and (in my opinion at least) not terribly helpful.

I have 2 ideas on implementation. The first borrowing an existing option for 'no-unused-vars':

```YML
no-unused-vars:
    - error
    - argsIgnorePattern: ^_ 
```

...or the existing 'allowedNames' option on the rule in question. Here's the proposed syntax:

```YML
  '@typescript-eslint/explicit-module-boundary-types':
      - error
      - allowedNames: ^[A-Z] 
# ...or
      - nameIgnorePattern: ^[A-Z] 

# or ^/[A-Z]/ - whatever works
```

**Expected Result**

So in a components you could go from this...

```TS
type CompProps = { prop: string }

const Comp = ({ prop }: CompProps): JSX.Element => {}
```
...to...
```TS
type CompProps = { prop: string }

const Comp = ({ prop }: CompProps) => {}
```

**Additional Info**

I'm admittedly still somewhat green when it comes to TS, so if this seems silly, my apologies! I only ask this because I'd still like to use return typing on functions/methods, and figured it may be useful for React and *possibly* other front-end frameworks that could similarly benefit.

**Comment by bradzacher** [comment]
This has been discussed ad nauseam - there will not be an ignore option added to either this rule or `explicit-function-return-types`.
